### PR TITLE
Add skeleton loader to HomeView (Play) page game list #512

### LINKED ISSUE #512: [Feature]: Add skeleton loader to HomeView (Play) page game list
### Feature Summary

When a user navigates to the HomeView (where we display the list of games available to play), we request the list from the server. We should display

### Detailed Description

We can define a `loading` state in the `HomeView`'s `data()`, default it to `false`, then set it to `true` in the `created()` hook before making the api call, then set it to `false` after the api call resolves. We could then render a `<v-skeleton-loader>` inside the list of games to show placeholders for the games that are being requested.

See the `StatsView` for an example of using a skeleton loader in Cuttle and the [vuetify docs](https://vuetifyjs.com/en/components/skeleton-loaders/) for full documentation on this component

### PR DESCRIPTION
**Pull request by kawaljen**
## 512

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- close #512

## Please check the following

- [ ] Do the tests still pass? 
- [ ] Is the code formatted properly? yes
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes? no
  - [ ] Has the documentation been updated accordingly? no

### CODE REVIEW

File: src/routes/home/HomeView.vue [hunk 1]
```diff
@@ -25,48 +25,80 @@
...
-                    :is-ranked="game.isRanked"
-                    :is-spectatable="true"
-                    :disable-spectate="game.isOver"
-                    @error="handleError"
-                  />
-                </div>
-              </v-window-item>
-            </v-window>
+            <template v-if="loadingData">
+              <div class="mx-3">
```
Comment (Haviles04) [inline src/routes/home/HomeView.vue:29]: I think we can probably just put the v-if on the div and remove the template

File: src/routes/home/HomeView.vue [hunk 2]
```diff
@@ -25,48 +25,80 @@
...
+                      type="heading"
+                      color="surface-2"
+                    />
+                  </v-col>
+                  <v-divider color="surface-1" class="border-opacity-25" />
+                </v-row>
+              </div>
+            </template>
+            <template v-else>
+              <v-window v-model="tab" class="pa-4 overflow-y-auto">
```
Comment (Haviles04) [inline src/routes/home/HomeView.vue:59]: same here

**Review by Haviles04 — COMMENTED** [review]
Nice work! I left a few nitpick suggestions, Could you also post a screenshot of how the loader looks?

File: src/routes/home/HomeView.vue (hunk 2 above)
Comment (kawaljen) [reply src/routes/home/HomeView.vue]: ok i think it's done

File: src/routes/home/HomeView.vue (hunk 1 above)
Comment (kawaljen) [reply src/routes/home/HomeView.vue]: Screenshot from 2023-11-29 10-40-16

Comment (kawaljen) [reply src/routes/home/HomeView.vue]: Screenshot from 2023-11-29 10-40-46

**Comment by kawaljen** [comment]
here are the screenshots of the loader
Screenshot from 2023-11-29 10-40-16
Screenshot from 2023-11-29 10-40-16

**Comment by kawaljen** [comment]
and on smaller device
Screenshot from 2023-11-29 10-40-46

**Pull request merged**